internal/provider/resource_project.go

```diff
@@ -14,10 +13,8 @@ import (
 	"github.com/kislerdm/terraform-provider-neon/internal/types"
 )
 
-const providerDefaultHistoryRetentionSeconds = int(time.Hour/time.Second) * 24 * 7
```

Comment:
**(suggestion, blocking)** I suggest to change the value to 1 day to work around the side effect which you explained [here](https://github.com/kislerdm/terraform-provider-neon/pull/124#issuecomment-2471975115). For context, it was my rationale behind having this value defined as default when the resource was added to the provider. 😀 Now, we are dealing with the negative consequence of that decision. However, I find having the risk of inconsistency with the Neon defaults is still preferable to maintain the behavior which may be expressed as:
> As the provider's user, I want the retention period changed to the default value when I remove that attribute from the `neon_project`, so I can be sure about my infra configuration.

WDYT?

Reply:
I think that's a fair choice 🙂 I wish there was a better way to handle that but I did think hard about it and couldn't come up with anything since ultimately at this point Neon doesn't have a way to retrieve what the default is.
That being said, I noticed that Neon's OpenAPI spec don't have the default value documented. I wonder if it was, if that would end up generate some constant we could import. Maybe someone more familiar than I am with OpenAPI would know 😄

Reply:
Done!

Reply:
@mickael-carl well done!

> Neon's OpenAPI spec don't have the default value documented

It's quite unfortunate indeed. If the config had default, we could have adjusted the Go SDK generation to maintain the defaults there, and use them in the tf provider and other, e.g. pulumi, providers / frameworks. Maybe it'd be worth bringing to the Neon API team 🤔